tests/components/recorder/test_purge.py

```diff
@@ -130,9 +130,16 @@ def test_purge_old_states(self):
             assert states.count() == 6
 
             # run purge_old_data()
-            purge_old_data(self.hass.data[DATA_INSTANCE], 4, repack=False)
+            finished = purge_old_data(self.hass.data[DATA_INSTANCE], 4, repack=False)
+            assert not finished
+            assert states.count() == 4
 
-            # we should only have 2 states left after purging
+            finished = purge_old_data(self.hass.data[DATA_INSTANCE], 4, repack=False)
+            assert not finished
+            assert states.count() == 2
+
+            finished = purge_old_data(self.hass.data[DATA_INSTANCE], 4, repack=False)
+            assert finished
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)